### ISSUE TITLE
Register form can't be submitted

### ISSUE THREAD

**Issue by r-martins** [ProxiBlue/reCaptcha#35]
**Steps to reproduce**
1. Select Create User on forms field
2. Using this config (but you can use invisible as well)
![configuration___system___administracao_magento](https://user-images.githubusercontent.com/191149/53606515-31d92080-3c0f-11e9-9e89-731b50be5ef7.jpg)

3. Go to account create page /customer/account/create/, fill the fields
4. Click on Submit button

**Expected Result**
Form is submitted or displays captcha challenge.

**Current Result**
Absolutely nothing happen. No messages on console, Network, var/log. Absolutely nothing.

**Additional info**
* If I try to submit the form with a required field missing, it displays the validation messages.
* Tried with Firefox and Chrome (with all extensions disabled)
* When disabling the module in /etc/modules, the Zend captcha appears, and it works.
* Magento 1.9.4.0 with porto theme (tested with rwd)
* Also tested with PHP 7.1 and 7.2
* Module version: 2.1.4 installed via modman
* If I type `$('form-validate').submit()` in console, it work as expected.
* If I use `<button type="submit"/>` or `<input type"submit"/>` it doesn't

**Comment by ProxiBlue** [comment]
Managed to reproduce, magento 1.9.4.0 only issue seems like it. (tried in 1.9.3.10, and worked fine)

You mention that "Absolutely nothing happen". This is not 100% what I see, so not 100% sue is same issue found.
What I get is that the form does submit, however flow is taken directly back to create page.

I can see this via the networking tab:

![selection_999 026](https://user-images.githubusercontent.com/4994260/53611442-909b9b80-3c09-11e9-8399-a0cd882ba9ee.png)

Looking at the code, the redirect is happening due to form_key invalidation

![selection_999 027](https://user-images.githubusercontent.com/4994260/53611468-ac06a680-3c09-11e9-89bb-af1591f7bd51.png)

**Comment by ProxiBlue** [comment]
Checking posted form data, there is a form key there...so why is it being rejected?

![selection_999 028](https://user-images.githubusercontent.com/4994260/53611563-00118b00-3c0a-11e9-8f29-5557d07ded6f.png)

**Comment by ProxiBlue** [comment]
Ok, so the form key in the posted form, and the one in the session differs. Seems like a stale form key problem.

![selection_999 029](https://user-images.githubusercontent.com/4994260/53611698-71513e00-3c0a-11e9-8fa3-7d62f7d00e0a.png)

**Comment by ProxiBlue** [comment]
mmm, ok, might be on a red-herring here, as I have disabled/removed the recapctha module, and the issue is persisting, which points to either I have a bad install, or this is not a bug related to recactha

**Comment by ProxiBlue** [comment]
@r-martins 

Does the problem persist if you disable reCapctha module (clear cache)
Can you reproduce on a fresh magento .9.2.4 install?

**Comment by ProxiBlue** [comment]
Ok, started over, brand new fresh install of 1.9.2.4, and registration now works. 
Re-installed reCaptcha using composer. Registration still works.

The previous issues (form key) found looked more like a magento upgrade issue (I upgraded 1.9.3.10 to 1.9.2.4) 
A fresh install of magento 1.9.2.4 fixed the registration.

Now, testing with magento 1.9.2.4, I cannot find any issue/reproduce issue.
This is possibly a theme issue.

Confirmation that issue remains / does not remains when module is removed would be great.
Possible also to check if you are getting teh same redirect issue I found initially. Maybe issue is with the form_key in the custom theme?

Video showing working for me:

https://drive.google.com/file/d/1zZvNwTbLtMygLtSc7dAlX5B_YP5_i8Zi/view

**Comment by r-martins** [comment]
Hi Lucas.
I can confirm that the problem doesn't happen when disabling the module.
See my video: https://www.youtube.com/watch?v=089irfSTwOI
I noticed you are in Australia as well. If you want we can chat somewhere else and debug it together.
I tried with rwd/default as well.

**Comment by ProxiBlue** [comment]
Do you have this running somewhere I can access?
I cannot reproduce, but it does seem like a javascript issue, so i could potentially figure out wtf is going on by seeing it in action, and debugging js code.

**Comment by r-martins** [comment]
Hi Lucas
I've installed the module on the public test environment, but wasn't able to reproduce the issue there.
Tomorrow I'll try to send the local db and unstaged settings there in order to reproduce.

Thanks for the support and help so far.

**Comment by r-martins** [comment]
Hi Lucas.
I'm starting to believe that this is something that I did, and not with the module itself.
When uploading this environment, I was only able to reproduce after uploading the working files (not added to the repo yet), but when just installing your module, it still works.

Anyway, kindly see if you are able to see any message or error there: http://teste.magenteiro.com/customer/account/create

**Comment by ProxiBlue** [comment]
Hi,

Looks like the issue might be the fact that you have commented out the varienForm creation.

![selection_999 048](https://user-images.githubusercontent.com/4994260/54003709-12517300-418e-11e9-997b-5d17385aab89.png)

The reCapctha code intercepts the varienForm submit, to effect the challenge

**Comment by r-martins** [comment]
Yeh, nice point that I left behind Lucas.
I just uncommented it...
It worked in the fist attempt.. but after refreshing and trying again, I got the same behaviour.

Thanks for helping me on this one.

**Comment by r-martins** [comment]
Hi Lucas.
I found the problem.
During this customization, I've changed one of my validation rules, and I got a JS function returning exception.
This makes the varien form validate() fail silently.

Sorry for the confusion. **This is not a bug on ProxiBlue module**. 🥇 

---
Further details (if you want):
js/ricardomartins/cpfvalidation.js

> function validateCPF(strCPF) {
>     countrySelect = document.querySelector("#billing\\:country_id");
>     if(countrySelect.value != 'BR'){
>         return true;
>     }

countrySelect is null on this form, so it gets an exception.
validateCPF is called in one of taxvat field validation rules.
